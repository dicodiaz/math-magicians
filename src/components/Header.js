import { AiOutlineMenu } from 'react-icons/ai';
import { Link } from 'react-router-dom';

const Header = () => (
  <nav className="navbar navbar-expand-md navbar-light bg-secondary fixed-top">
    <div className="container-fluid">
      <span className="navbar-brand text-white">Math Magicians</span>
      <button
        className="navbar-toggler"
        type="button"
        data-bs-toggle="collapse"
        data-bs-target="#navbarNav"
        aria-controls="navbarNav"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <AiOutlineMenu style={{ color: 'white', fontSize: '25px' }} />
      </button>
      <div className="collapse navbar-collapse" id="navbarNav">
        <ul className="navbar-nav ms-auto">
          <li className="nav-item">
            <Link to="/">Home</Link>
          </li>
          <li className="v-line d-none d-md-block" />
          <li className="nav-item">
            <Link to="/calculator">Calculator</Link>
          </li>
          <li className="v-line d-none d-md-block" />
          <li className="nav-item">
            <Link to="/quote">Quote</Link>
          </li>
        </ul>
      </div>
    </div>
  </nav>
);

export default Header;
