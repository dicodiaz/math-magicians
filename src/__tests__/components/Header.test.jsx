import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import Header from '../../components/Header';

describe('Header component', () => {
  it('should render as expected', () => {
    const { container } = render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
    );
    expect(container).toMatchSnapshot();
  });
});
