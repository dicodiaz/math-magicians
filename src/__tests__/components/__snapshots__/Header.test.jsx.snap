// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render as expected 1`] = `
<div>
  <nav
    class="navbar navbar-expand-md navbar-light bg-secondary fixed-top"
  >
    <div
      class="container-fluid"
    >
      <span
        class="navbar-brand text-white"
      >
        Math Magicians
      </span>
      <button
        aria-controls="navbarNav"
        aria-expanded="false"
        aria-label="Toggle navigation"
        class="navbar-toggler"
        data-bs-target="#navbarNav"
        data-bs-toggle="collapse"
        type="button"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="color: white; font-size: 25px;"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
          />
        </svg>
      </button>
      <div
        class="collapse navbar-collapse"
        id="navbarNav"
      >
        <ul
          class="navbar-nav ms-auto"
        >
          <li
            class="nav-item"
          >
            <a
              href="/"
            >
              Home
            </a>
          </li>
          <li
            class="v-line d-none d-md-block"
          />
          <li
            class="nav-item"
          >
            <a
              href="/calculate"
            >
              Calculator
            </a>
          </li>
          <li
            class="v-line d-none d-md-block"
          />
          <li
            class="nav-item"
          >
            <a
              href="/quote"
            >
              Quote
            </a>
          </li>
        </ul>
      </div>
    </div>
  </nav>
</div>
`;
